src/main/java/edu/ucsb/cs156/gauchoride/config/SecurityConfig.java

```diff
@@ -88,6 +88,13 @@ private GrantedAuthoritiesMapper userAuthoritiesMapper() {
           if (email.endsWith("@ucsb.edu")) {
             mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_MEMBER"));
           }
+
+          Optional<User> u = userRepository.findByEmail(email);
+          if (u.isPresent() && u.get().getDriver()) {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_DRIVER"));
+          } else {
+            mappedAuthorities.add(new SimpleGrantedAuthority("ROLE_RIDER"));
```

Comment:
This should also be in it's own PR; it's a separate concern to establish the role from adding the backend for driver shifts.

It's also in your best interest to have many smaller PRs... you'll earn more points that way.

I deliberately incentivize that because it leads you to the habits that folks in industry want: small, focused PRs, not big sprawling hard to review ones.